asteroid/foo.py

```diff
@@ -0,0 +1,162 @@
+import os
+import torch
+import numpy as np
+import warnings
+
+try:
+    from typing import Protocol
+except ImportError:  # noqa
+    # Python < 3.8
+    class Protocol:
+        pass
+
+
+from asteroid.utils import get_device
+
+
+class Separatable(Protocol):
+    """Things that are separatable.
```

Comment:
Oh think "separatable" would be more common, but from a quick Google it seems the more appropriate term would be "separable". Shall we change to that?

Reply:
@mpariente

Reply:
I thought that this was a naming convention. That because the method was `separate`, we called it `Separatable`. Isn't this the case?

Reply:
Yes but seems like to proper name would be "separable".

We could also call the interface `Separate`, which is common eg. in Rust (eg. https://doc.rust-lang.org/reference/special-types-and-traits.html#copy instead of Copyable).

Let's just pick any name, I don't really care :D